server/src/main/java/org/cloudfoundry/identity/uaa/provider/saml/UserAttributeChangesSnsHandler.java

```diff
@@ -0,0 +1,264 @@
+package org.cloudfoundry.identity.uaa.provider.saml;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ge.iam.sns.service.MessageBuilder;
+import com.ge.iam.sns.service.SnsService;
+import com.google.gson.JsonObject;
+import org.apache.commons.lang.StringUtils;
+import org.cloudfoundry.identity.uaa.events.UserAttributeChangedEvent;
+import org.cloudfoundry.identity.uaa.provider.IdentityProvider;
+import org.cloudfoundry.identity.uaa.provider.IdentityProviderProvisioning;
+import org.cloudfoundry.identity.uaa.user.UaaUser;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.scheduling.annotation.EnableAsync;
+import org.springframework.stereotype.Component;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Component
+@EnableAsync
+public class UserAttributeChangesSnsHandler {
+
+    private final static Logger logger = LoggerFactory.getLogger(UserAttributeChangesSnsHandler.class);
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Value("${sns.topic.arn:arn}")
+    private String snsTopicArn;
+
+    @Value("${sns.enabled:true}")
+    private boolean snsEnabled;
+
+    private final SnsService snsService;
+    private final IdentityProviderProvisioning identityProviderProvisioning;
+
+    @Autowired
+    public UserAttributeChangesSnsHandler(SnsService snsService,
+            IdentityProviderProvisioning identityProviderProvisioning) {
+        if (snsService == null) {
+            throw new IllegalArgumentException("SnsService cannot be null");
```

Comment:
This is redundant code, if bean is not available, spring will automatically throw exception. This will be helpful only if you want to have some exception thrown, when you will manually instantiate the class in the test. I think, mostly not required.